drivers/stepper/step_dir/step_dir_stepper_counter_timing.c

```diff
@@ -0,0 +1,108 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 Fabian Blatz <fabianblatz@gmail.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/counter.h>
+#include "step_dir_stepper_common.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(step_dir_stepper);
+
+static void step_counter_top_interrupt(const struct device *dev, void *user_data)
+{
+	ARG_UNUSED(dev);
+	struct step_dir_stepper_common_data *data = user_data;
+
+	stepper_handle_timing_signal(data->dev);
+}
+
+int step_counter_timing_source_update(const struct device *dev, const uint32_t velocity)
+{
+	const struct step_dir_stepper_common_config *config = dev->config;
+	struct step_dir_stepper_common_data *data = dev->data;
+	int ret;
+
+	if (velocity == 0) {
+		return -EINVAL;
+	}
+
+	data->counter_top_cfg.ticks =
+		DIV_ROUND_UP(counter_us_to_ticks(config->counter, USEC_PER_SEC), velocity);
+
+	ret = counter_set_top_value(config->counter, &data->counter_top_cfg);
+	if (ret != 0) {
+		LOG_ERR("%s: Failed to set counter top value (error: %d)", dev->name, ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+int step_counter_timing_source_start(const struct device *dev)
+{
+	const struct step_dir_stepper_common_config *config = dev->config;
+	int ret;
+
+	ret = counter_start(config->counter);
+	if (ret < 0) {
+		LOG_ERR("Failed to start counter: %d", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+int step_counter_timing_source_stop(const struct device *dev)
+{
+	const struct step_dir_stepper_common_config *config = dev->config;
+	int ret;
+
+	ret = counter_stop(config->counter);
+	if (ret < 0) {
+		LOG_ERR("Failed to stop counter: %d", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+bool step_counter_timing_source_needs_reschedule(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+	return false;
+}
+
+bool step_counter_timing_source_is_running(const struct device *dev)
+{
+	const struct step_dir_stepper_common_config *config = dev->config;
+
+	return counter_is_counting_up(config->counter);
```

Comment:
Rereading the diff: Not sure if that is that check is the right one. This seems to be a static property and not like "is the counter currently runnig".

Reply:
Yeah, the drv8424 driver used a seperate bool for tracking this. The method is indeed not an indicator if the counter is "active" or not. Adopted the approach of the existing driver instead.